subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/vcs/VcsDependencyResolver.java

```diff
@@ -122,6 +129,13 @@ private File populateWorkingDirectory(File baseWorkingDir, VersionControlSpec sp
         return versionControlSystem.populate(versionDirectory, selectedVersion, spec);
     }
 
+    private VersionRef selectVersionFromCache(VersionControlSpec spec, String version) {
+        if (VersionRef.LATEST_INTEGRATION.equals(version)) {
```

Comment:
Despite the phrasing of the issue. There is nothing special about `latest.integration` for this functionality. For any version we get, we should only get it once during the build. So, say the version is `master` or `development`. We'd want to get the version from the version control system for those branches as well, but still only the first time we try to do the resolve during a build.

Reply:
Good point, changed.